### PR TITLE
Rename data entity fields according to ODD specification

### LINKED ISSUE #277: Use the same naming in odd specification and odd platform
e.g. in specification we have DataEntityType, which is mapped to DataEntitySubType in platform

### PR DESCRIPTION
**Pull request by alinamiryuk**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
- "type" naming were changed to "entityClass"
- "subtype" naming were changed to "type"

**Is there anything you'd like reviewers to focus on?**
Resolved #277  

**How Has This Been Tested?** (put an "X" next to an item)

- [X] Manually(please, describe, when necessary)

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code

Check out [Contributing](https://github.com/opendatadiscovery/odd-platform/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/opendatadiscovery/odd-platform/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: odd-platform-ui/src/components/Alerts/AlertsList/AlertItem/AlertItem.tsx [hunk 1]
```diff
@@ -53,11 +53,11 @@ const AlertItem: React.FC<AlertItemProps> = ({
...
       <S.TypesContainer>
-        {alert.dataEntity?.types?.map(type => (
-          <EntityTypeItem
+        {alert.dataEntity?.entityClasses?.map(entityClass => (
+          <EntityClassItem
             sx={{ ml: 0.5 }}
-            key={type.id}
-            typeName={type.name}
+            key={entityClass.id}
+            typeName={entityClass.name}
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Alerts/AlertsList/AlertItem/AlertItem.tsx:60]: `entityClassName={entityClass.name}`

Comment (Leshe4ka) [inline odd-platform-ui/src/components/Alerts/AlertsList/AlertItem/AlertItem.tsx:60]: > entityClassName={entityClass.name}

File: odd-platform-ui/src/redux/interfaces/state.ts [hunk 2]
```diff
@@ -158,7 +158,7 @@ export interface DataEntitiesState {
   popular: DataEntityRef[];
   typesDict: {
     types: Dictionary<DataEntityType>;
-    subtypes: Dictionary<DataEntitySubType>;
+    entityClasses: Dictionary<DataEntityClass>;
   };
```
Comment (Leshe4ka) [inline odd-platform-ui/src/redux/interfaces/state.ts:162]: I think that now it's not only `typesDict`, but `classAndTypeDict`.
And I suggest using the same naming - `entityTypes` and `entityClasses`.

File: odd-platform-ui/src/components/shared/AppGraph/AppGraphNode/AppGraphNode.tsx [hunk 3]
```diff
@@ -350,15 +350,15 @@ const AppGraphNode: React.FC<AppGraphNodeProps> = ({
...
             transform={`translate(${
               titleLayout.x + i * (typeLayout.width + typeLayout.mx)
             },${nodeSize.y - typeLayout.my - typeLayout.height})`}
           >
             <TypeContainer
-              $typeName={type.name}
+              $typeName={entityClass.name}
               width={typeLayout.width}
               height={typeLayout.height}
             />
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/shared/AppGraph/AppGraphNode/AppGraphNode.tsx:364]: Now it's `EntityClassContainer` with `$entityClassName` prop

File: odd-platform-ui/src/components/shared/EntityClassItem/EntityClassItemStyles.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import styled from 'styled-components';
-import { DataEntityTypeNameEnum } from 'generated-sources';
+import { DataEntityClassNameEnum } from 'generated-sources';
 import { Box } from '@mui/material';
 
 export const Content = styled(Box)<{
-  typeName: DataEntityTypeNameEnum;
+  typeName: DataEntityClassNameEnum;
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/shared/EntityClassItem/EntityClassItemStyles.ts:6]:
```suggestion
  entityClassName: DataEntityClassNameEnum;
```

and add $ prefix for styled-component props, please

File: odd-platform-ui/src/redux/actions/dataentity.actions.ts [hunk 5]
```diff
@@ -4,16 +4,16 @@ import {
...
 } from 'generated-sources';
 import { PartialEntityUpdateParams, ErrorState } from 'redux/interfaces';
 
-export const fetchDataEntitiesTypesAction = createAsyncAction(
+export const fetchDataEntitiesClassesAndTypesAction = createAsyncAction(
   'GET_DATA_ENTITIES_TYPES__REQUEST',
   'GET_DATA_ENTITIES_TYPES__SUCCESS',
   'GET_DATA_ENTITIES_TYPES__FAILURE'
-)<undefined, DataEntityTypeDictionary, ErrorState>();
+)<undefined, DataEntityClassAndTypeDictionary, ErrorState>();
```
Comment (Leshe4ka) [inline odd-platform-ui/src/redux/actions/dataentity.actions.ts:16]: `'GET_DATA_ENTITIES_CLASSES_AND_TYPES__REQUEST'`, etc.

File: odd-platform-ui/src/redux/selectors/dataentity.selectors.ts [hunk 6]
```diff
@@ -211,16 +211,16 @@ export const getDataEntityIsDataset = createSelector(
...
 );
 
 export const getDataEntityIsQualityTest = createSelector(
   dataEntitiesState,
   getDataEntityId,
   (dataEntities, dataEntityId) =>
-    !!dataEntities.byId[dataEntityId]?.types.find(
-      type => type.name === DataEntityTypeNameEnum.QUALITY_TEST
+    !!dataEntities.byId[dataEntityId]?.entityClasses?.find(
+      type => type.name === DataEntityClassNameEnum.QUALITY_TEST
```
Comment (Leshe4ka) [inline odd-platform-ui/src/redux/selectors/dataentity.selectors.ts:224]:
```suggestion
      class => class.name === DataEntityClassNameEnum.QUALITY_TEST
```

File: odd-platform-ui/src/redux/selectors/dataentity.selectors.ts [hunk 7]
```diff
@@ -211,16 +211,16 @@ export const getDataEntityIsDataset = createSelector(
   dataEntitiesState,
   getDataEntityId,
   (dataEntities, dataEntityId) =>
-    !!dataEntities.byId[dataEntityId]?.types.find(
-      type => type.name === DataEntityTypeNameEnum.SET
+    !!dataEntities.byId[dataEntityId]?.entityClasses?.find(
+      type => type.name === DataEntityClassNameEnum.SET
```
Comment (Leshe4ka) [inline odd-platform-ui/src/redux/selectors/dataentity.selectors.ts:215]:
```suggestion
      class => class.name === DataEntityClassNameEnum.SET
```

File: odd-platform-ui/src/redux/reducers/dataentitiesSearch.reducer.ts [hunk 8]
```diff
@@ -174,7 +174,8 @@ const clearFilters = (state: SearchState): SearchState => ({
   ...state,
   isFacetsStateSynced: false,
   facetState: mapValues(state.facetState, (filter, facetName) => {
-    if (facetName === 'types') return state.facetState.types; // Not clearing types filter
+    if (facetName === 'entityClasses')
+      return state.facetState.entityClasses; // Not clearing types filter
```
Comment (Leshe4ka) [inline odd-platform-ui/src/redux/reducers/dataentitiesSearch.reducer.ts:178]: Refactor or delete comment, please

**Review by Leshe4ka — CHANGES_REQUESTED** [review]

File: odd-platform-ui/src/redux/reducers/dataentitiesSearch.reducer.ts (hunk 8 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/reducers/dataentitiesSearch.reducer.ts:178]: done

File: odd-platform-ui/src/redux/selectors/dataentity.selectors.ts (hunk 7 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/selectors/dataentity.selectors.ts:215]: done

File: odd-platform-ui/src/redux/selectors/dataentity.selectors.ts (hunk 6 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/selectors/dataentity.selectors.ts:224]: done

File: odd-platform-ui/src/redux/actions/dataentity.actions.ts (hunk 5 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/actions/dataentity.actions.ts:16]: done

File: odd-platform-ui/src/components/shared/EntityClassItem/EntityClassItemStyles.ts (hunk 4 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/shared/EntityClassItem/EntityClassItemStyles.ts:6]: done

File: odd-platform-ui/src/components/shared/AppGraph/AppGraphNode/AppGraphNode.tsx (hunk 3 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/shared/AppGraph/AppGraphNode/AppGraphNode.tsx:364]: done

File: odd-platform-ui/src/redux/interfaces/state.ts (hunk 2 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/interfaces/state.ts:162]: done

File: odd-platform-ui/src/redux/selectors/dataentitySearch.selectors.ts [hunk 9]
```diff
@@ -111,13 +111,13 @@ export const getSearchFacetsByType = createSelector(
   (search, searchFacet) => values(search.facets[searchFacet]?.items) || []
 );
 
-export const getSearchEntityType = createSelector(searchState, search => {
+export const getSearchEntityClass = createSelector(searchState, search => {
   if (search.myObjects) return 'my' as SearchType;
```
[the remark this answers is not in the document]
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/selectors/dataentitySearch.selectors.ts:115]: done

File: odd-platform-ui/src/components/shared/EntityClassItem/EntityClassItem.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { DataEntityClassNameEnum } from 'generated-sources';
+import { DataEntityClassLabelMap } from 'redux/interfaces/dataentities';
+import { BoxProps } from '@mui/material';
+import * as S from './EntityClassItemStyles';
+
+export interface EntityClassItemProps extends Pick<BoxProps, 'sx'> {
+  typeName: DataEntityClassNameEnum;
```
[the remark this answers is not in the document]
Comment (alinamiryuk) [reply odd-platform-ui/src/components/shared/EntityClassItem/EntityClassItem.tsx:8]: done

**Review by DementevNikita — APPROVED** [review]

**Review by Leshe4ka — APPROVED** [review]

**Pull request merged**